### PR TITLE
fix: resolve type mismatch in allServers and allChannels

### LINKED ISSUE #853: `.all()` method doesn't work as expected for `allServers` and `allChannels`
#### Describe the bug
I am trying to get an iterable list of different components from the asyncapi document. by calling `allMessages` I get a `Collection` of Messages that I can convert to an array by calling `.all()` method.

this approach works as expected for:
allMessages()
allSchemas()
allOperations()

but not for the following components:
allServers()
allChannels()

#### Expected behavior
I expect the `.all()` method to behave the same way for all of the components.

### PR DESCRIPTION
**Pull request by KhudaDad414**
fixes #853

### CODE REVIEW

**Comment by KhudaDad414** [comment]
@smoya Also added some tests for v3 of the same methods.

File: src/models/v3/asyncapi.ts [hunk 1]
```diff
@@ -106,32 +106,31 @@ export class AsyncAPIDocument extends BaseModel<v3.AsyncAPIObject> implements As
...
     return new Channels(channels);
   }
 
   allOperations(): OperationsInterface {
-    const operations: OperationInterface[] = [];
-    this.allChannels().forEach(channel => operations.push(...channel.operations()));
+    const operations: OperationInterface[] = this.operations().all();
+    this.components().operations().forEach(operation => 
+      !operations.some(o => o.json() === operation.json()) && operations.push(operation)
+    );
```
Comment (smoya) [inline src/models/v3/asyncapi.ts:128]: 💯 💯

File: src/models/v3/asyncapi.ts [hunk 2]
```diff
@@ -106,32 +106,31 @@ export class AsyncAPIDocument extends BaseModel<v3.AsyncAPIObject> implements As
...
     return new Operations(operations);
   }
 
   allMessages(): MessagesInterface {
-    const messages: MessageInterface[] = [];
-    this.allOperations().forEach(operation => operation.messages().forEach(message => (
-      !messages.some(m => m.json() === message.json()) && messages.push(message)
-    )));
+    const messages: MessageInterface[] = this.messages().all();
```
Comment (smoya) [inline src/models/v3/asyncapi.ts:133]: 💯 💯

**Review by smoya — APPROVED** [review]
LGTM! 🚀🌔

Thank you for this super bug fix and for covering this up with tests 🙌

**Pull request merged**